pkg/provider/vault/vault.go

```diff
@@ -816,41 +855,57 @@ func getCertFromConfigMap(v *client) ([]byte, error) {
 	return []byte(val), nil
 }
 
+/*
+setAuth gets a new token using the configured mechanism.
+If there's already a valid token, does nothing.
+*/
 func (v *client) setAuth(ctx context.Context, cfg *vault.Config) error {
-	tokenExists, err := setSecretKeyToken(ctx, v)
+	tokenExists, err := checkToken(ctx, v.token)
+	if tokenExists {
+		v.log.V(1).Info("Re-using existing token")
+		return err
+	}
```

Comment:
The e2e tests on main seem to be fine. only a particular auth method (cert) is failing, everything is is good.

I assume this `checkToken` causes the cert-based authentication to fail. It seems that when using the token from the `cert/auth/login` requires the use of TLS with a client certificate. Tho i couldn't find that detail in the documentation :raised_eyebrow:.

Also, calling `.lock()` before `.initialize()` gives a hard error due `c.cache` and `c.mu` being nil.